pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -409,8 +409,13 @@ func (c *Controller) InstancesByPort(hostname model.Hostname, reqSvcPort int,
 
 // GetProxyServiceInstances returns service instances co-located with a given proxy
 func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.ServiceInstance, error) {
-	var out []*model.ServiceInstance
 	proxyIP := proxy.IPAddress
+	podNamespace := ""
+	if pod, exists := c.pods.getPodByIP(proxyIP); exists {
+		podNamespace = pod.Namespace
+	}
+	endpointsForPodInSameNS := make([]*model.ServiceInstance, 0)
+	endpointsForPodInDifferentNS := make([]*model.ServiceInstance, 0)
 	for _, item := range c.endpoints.informer.GetStore().List() {
```

Comment:
BTW, not relate to this pr.
 I have another doubt: each proxy we do iterate on all endpoints and services, is that efficient? Can we do some optimization？